src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -3867,5 +3853,21 @@ private static SocketError GetSocketErrorFromFaultedTask(Task t)
                 _ => SocketError.SocketError
             };
         }
+
+        private void CheckNonBlockingConnectCompleted()
+        {
+            if (_nonBlockingConnectInProgress)
+            {
+                if (SocketPal.HasNonBlockingConnectCompleted(_handle, out bool success))
```

Comment:
Nit: I'm not sure if we have guideline points about this, but I would remove the double nesting:
```C#
if (_nonBlockingConnectInProgress & SocketPal.HasNonBlockingConnectCompleted(_handle, out bool success))
```